src/implicit_restart.jl

```diff
@@ -156,13 +156,16 @@ function double_shift!(H_whole::AbstractMatrix{Tv}, min, max, μ::Complex, Q::Ab
         mul!(Q, G₂)
     end
 
-    # Do the last Given's rotation by hand.
-    H[n - 1, n - 2] = H[n + 1, n - 2]
+    if n > 2
```

Comment:
Why is this necessary btw?

Reply:
If the active part is only 2 elements wide, then we'd be indexing at [1,0] or [3,0] (it errored once because of this). I'm not sure of just adding "if n > 2" does the exact right thing, but it passed the tests

Reply:
Right, it makes sense. Maybe it also makes sense to enforce |max - min| > something.